### ISSUE TITLE
pip docker-compose doesn't seems to have the tag "latest" anymore

### ISSUE THREAD

**Issue by Platou** [nickjj/ansible-docker#47]
Installing the role: 

```yml
  roles:
  - { role: "nickjj.docker", tags: ["docker"] }
```

Output:

```
TASK [nickjj.docker : Install Python packages] *********************************
ok: [host] => (item={u'state': u'present', u'name': u'docker'})
failed: [host] (item={u'path': u'/usr/local/bin/docker-compose', u'state': u'present', u'version': u'latest', u'name': u'docker-compose', u'src': u'/usr/local/lib/docker/virtualenv/bin/docker-compose'}) => {"changed": false, "cmd": "/usr/local/lib/docker/virtualenv/bin/pip2 install docker-compose==latest", "item": {"name": "docker-compose", "path": "/usr/local/bin/docker-compose", "src": "/usr/local/lib/docker/virtualenv/bin/docker-compose", "state": "present", "version": "latest"}, "msg": "stdout: Collecting docker-compose==latest\n\n:stderr:   Could not find a version that satisfies the requirement docker-compose==latest (from versions: 1.1.0rc1, 1.1.0rc2, 1.1.0, 1.2.0rc1, 1.2.0rc2, 1.2.0rc3, 1.2.0rc4, 1.2.0, 1.3.0rc1, 1.3.0rc2, 1.3.0rc3, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.4.0rc1, 1.4.0rc2, 1.4.0rc3, 1.4.0, 1.4.1, 1.4.2, 1.5.0rc1, 1.5.0rc2, 1.5.0rc3, 1.5.0, 1.5.1, 1.5.2, 1.6.0rc1, 1.6.0, 1.6.1, 1.6.2, 1.7.0rc1, 1.7.0rc2, 1.7.0, 1.7.1, 1.8.0rc1, 1.8.0rc2, 1.8.0, 1.8.1, 1.9.0rc1, 1.9.0rc2, 1.9.0rc3, 1.9.0rc4, 1.9.0, 1.10.0rc1, 1.10.0rc2, 1.10.0, 1.10.1, 1.11.0rc1, 1.11.0, 1.11.1, 1.11.2, 1.12.0rc1, 1.12.0rc2, 1.12.0, 1.13.0rc1, 1.13.0, 1.14.0rc1, 1.14.0rc2, 1.14.0, 1.15.0rc1, 1.15.0, 1.16.0rc1, 1.16.0rc2, 1.16.0, 1.16.1, 1.17.0rc1, 1.17.0, 1.17.1, 1.18.0rc1, 1.18.0rc2, 1.18.0, 1.19.0rc1, 1.19.0rc2, 1.19.0rc3, 1.19.0, 1.20.0rc1, 1.20.0rc2, 1.20.0, 1.20.1, 1.21.0rc1, 1.21.0, 1.21.1, 1.21.2, 1.22.0rc1, 1.22.0rc2, 1.22.0, 1.23.0rc1, 1.23.0rc2, 1.23.0rc3, 1.23.0, 1.23.1, 1.23.2)\nNo matching distribution found for docker-compose==latest\n"}
```

**Comment by Platou** [comment]
Microsoft Azure : Canonical, UbuntuServer, 18.04-LTS, latest

**Comment by nickjj** [comment]
Here's what I used for testing where everything ran successfully:

- Set up those 3 distros using Docker images
- Created a stock Ubuntu 18.04 VM using the server ISO on Ubuntu's site
- Set up a brand new DigitalOcean 18.04 server

In all 3 cases, the role ran to completion and installed the latest version of Docker and Docker Compose.

In all 3 cases, I ran `apt-get update -y && apt-get install -y python` before this role since that's required for Ansible to run. This is a common bootstrap strategy.

What happens if you add that to a `pre_task` in your playbook?

**Comment by nickjj** [comment]
Here's a complete task you can drop in:

```yml
- name: Install Python without using Ansible modules
  raw: >
    bash -c "test -e /usr/bin/python \
    || (apt-get -qy update && apt-get install -qy python)"
  changed_when: false
```

It's using raw since this is meant to be ran before Ansible is even capable of being run.

**Comment by Platou** [comment]
This is my playbook
```
- hosts: "host"
  become: true
  become_method: sudo

  roles:
  - { role: "nickjj.docker", tags: ["docker"] }

  pre_tasks:
  - name: Install python 2
    raw: test -e /usr/bin/python || (apt -y update && apt install -y python-minimal)
```

**Comment by Platou** [comment]
Same error with `python` instead of `python-minimal`

**Comment by nickjj** [comment]
You could try adding this after the `- name: Install Virtualenv` task:

```yml
- name: Upgrade PIP
  pip:
    name: "pip"
    virtualenv: "{{ docker__pip_virtualenv }}"
    state: "forcereinstall"
```

Truthfully I'm not sure why that would be needed on Azure, but it's worth trying.

**Comment by nickjj** [comment]
Also what version of Ansible are you using? Maybe the `pip` package changed in some way to automatically convert `latest` to being omit. I'm running `2.7.5`.

**Comment by nickjj** [comment]
Another thing you could try is setting `docker__compose_version: ""`. Let me know if that installs the latest version without erroring out. If it does, that's likely something I should merge into the next release.

**Comment by Platou** [comment]
I'm using Ansible 2.5.1

**Comment by Platou** [comment]
Working with Ansible 2.7.5

**Comment by nickjj** [comment]
Thanks. Did you happen to try `""` with 2.5.1?

**Comment by nickjj** [comment]
Closing this. It's been addressed in `v1.8.0` which is available in the Galaxy. The role is back to working with Ansible 2.5 and 2.7.
